src/test/ui/rfc-2632-const-trait-impl/const-impl-trait.rs

```diff
@@ -0,0 +1,28 @@
+// check-pass
+#![feature(associated_type_bounds, const_trait_impl, const_cmp)]
+
+use std::marker::Destruct;
+
+const fn cmp(a: &impl ~const PartialEq) -> bool {
+    a == a
+}
+
+const _: () = {
+    assert!(cmp(&0xDEADBEEFu32));
+    assert!(cmp(&()));
+};
+
+#[const_trait]
+trait T {}
+struct S;
+impl const T for S {}
+
+const fn rpit() -> impl ~const T { S }
+
+const fn apit(_: impl ~const T + ~const Destruct) {}
+
+const fn rpit_assoc_bound() -> impl IntoIterator<Item: ~const T> { Some(S) }
+
+const fn apit_assoc_bound(_: impl IntoIterator<Item: ~const T> + ~const Destruct) {}
```

Comment:
please also test something like these (maybe returning an `impl PartialEq` and comparing at the call site?

Reply:
done.